pkg/proto/aksnodeconfig/v1/tlsbootstrappingconfig.proto

```diff
@@ -1,13 +1,26 @@
 syntax = "proto3";
 package aksnodeconfig.v1;
- 
+
+enum BootstrappingAuthMethod  {
+  //nolint:gosec // this is a const string to use in switch statements, not hardcoded credentials
+  UseTLSBootstrapToken = 0;
+  UseSecureTLSBootstrapping = 1;
+  UseArcMsiToMakeCSR = 2;
+  UseAzureMsiToMakeCSR = 3;
+  UseArcMsiDirectly = 4;
+  UseAzureMsiDirectly = 5;
+}
+
 message TLSBootstrappingConfig {
   // Enable secure TLS bootstrapping for the node.
-  optional bool enable_secure_tls_bootstrapping = 1;
+  BootstrappingAuthMethod bootstrapping_auth_method = 1;
 
   // Only required until Secure TLS bootstrapping in place. Would use kubelet identity after that.
-  string tls_bootstrapping_token = 2; 
+  string tls_bootstrapping_token = 2;
+
+  // Only used when secure TLS bootstrapping is enabled or one of the Azure/Arc methods. This is the appserver appid that the node will use to bootstrap.
+  string custom_aad_appserver_appid = 3;
```

Comment:
while you're at it please rename to `custom_aad_resource`, I originally planned on doing this in one of my previous PRs that has yet to merge